src/mongo/dbtests/dbtests.cpp

```diff
@@ -156,6 +158,10 @@ int dbtestsMain(int argc, char** argv, char** envp) {
         ->setFollowerMode(repl::MemberState::RS_PRIMARY)
         .ignore();
 
+    repl::DropPendingCollectionReaper::set(
```

Comment:
This will leak an instance of StorageInterfaceMock. You have to set the singleton using StorageInterface::set(). Here's an example (change StorageInterfaceImpl to StorageInterfaceMock):

https://github.com/mongodb/mongo/blob/523531464b747daa2a25eea33ed42239db871622/src/mongo/dbtests/storage_timestamp_tests.cpp#L91

Reply:
yes, i know, i just noticed that this is a ```main``` function, plus code above in this function doesn't care about similar leakage. Ok let me fix that. Thanks

Reply:
Are you referring to the call to "new repl::ReplicationCoordinatorMock(service, replSettings)"? repl::setGlobalReplicationCoordinator() creates a std::unique_ptr to manage the memory of the ReplicationCoordinator instance passed in.

Reply:
@banker right, but wan't it leak in some specific case when ReplicationCoordinator is instantiated and unique pointer was unable to be constructed for some reason? Strictly speaking instance of ReplicationCoordinator spawns before unique pointer gets a control. I'm not a big C++ expert yet, but i'm trying to follow up :D